src/main/webapp/app/overview/course-overview.component.ts

```diff
@@ -194,6 +204,94 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
+        this.updateMenuOffset();
+    }
+
+    /** Listen click event whether on outside of the menu or one of the items in the menu to close the dropdown menu */
+    @HostListener('document: click', ['$event'])
+    onClickCloseDropdownMenu() {
+        if (this.dropdownOpen) {
+            this.dropdownClickNumber += 1;
+            if (this.dropdownClickNumber === 2) {
+                this.dropdownOpen = false;
+                this.dropdownClickNumber = 0;
+            }
+        }
+    }
+
+    /** Update sidebar item's hidden property based on the window height to display three-dots */
+    updateVisibility(height: number) {
+        const thresholds: number[] = [];
+        const itemHeight: number = 35;
+        let threshold = this.calculateThreshold();
+        this.anyItemHidden = false;
+        this.hiddenItems = [];
+
+        for (let i = 0; i < this.sidebarItems.length - 1; i++) {
+            thresholds.unshift(threshold);
+            threshold -= itemHeight;
+        }
+        thresholds.unshift(0);
+
+        this.sidebarItems.forEach((item, index) => {
+            item.hidden = height <= thresholds[index];
+            if (item.hidden) {
+                this.anyItemHidden = true;
+                this.hiddenItems.push(item);
+            }
+        });
+    }
+
+    /** Calculate dropdown-menu position based on the number of entries in the sidebar */
+    updateMenuOffset() {
+        const leftSidebarItems: number = this.sidebarItems.length - this.hiddenItems.length;
+        const minHeightForSidebar: number = 330; // Minimum height for the sidebar to dynamically position dropdown menu
+        const offsetForEachElement: number = 40; // Offset to be added to the dropdown menu for each element left in the sidebar
+        const maxSidebarElements: number = 9;
+
+        if (leftSidebarItems === 1) {
+            // In order to fit all content of the dropdown menu, we put the menu at the top of the page
+            this.dropdownOffset = 0;
+            if (window.innerHeight > minHeightForSidebar) {
+                // To move the dropdown menu a little to the bottom even if there is one element left in the sidebar but the minimum height of the sidebar is not exceeded yet
+                this.dropdownOffset += (maxSidebarElements - this.hiddenItems.length) * 10;
+            }
+        } else {
+            // To adjust dropdown menu position based on the amount of entries/elements left in the sidebar
+            this.dropdownOffset = leftSidebarItems * offsetForEachElement;
+        }
+    }
+
+    /** Calculating threshold levels based on the number of entries in the sidebar */
+    calculateThreshold(): number {
+        const maxThreshold: number = 650; // The threshold level at which the 'More' element will appear
+        const offsetForEachElement: number = 35; // Offset to be subtracted from the max threshold level for each element in the sidebar
+        const maxSidebarElements: number = 9;
```

Comment:
I would suggest uppercase and defining the variables at the top of this component aswell.

Please also reuse the variables that are already introduced, `offsetForEachElement` and `maxSidebarElements` seems to be the exact same variable that is locally defined in `updateMenuOffset` aswell